lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -56,25 +56,103 @@ define((require, exports, module) ->
     jqEditorContainer = $(editor.container)
     getFormulaElement = -> $("#formula")
     KATEX_OPTIONS = {displayMode: true, throwOnError: false}
+    TokenIterator = require("ace/token_iterator").TokenIterator
+    Range = require("ace/range").Range
+
+    erh = EquationRangeHandler = {
+      BEGIN_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\begin"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+      END_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\end"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+      rangeCache: {}
+
+      compareTokens: (token1, token2) ->
+        token1["type"] == token2["type"] and token1["value"] == token2["value"]
```

Comment:
Yes, please change it. We work with tokens like objects in other parts of code and Ace itself treats them as objects as well